lib/matplotlib/cbook/__init__.py

```diff
@@ -332,9 +331,7 @@ def __setstate__(self, state):
         self.__init__(**state)
 
     def connect(self, s, func):
-        """
-        register *func* to be called when a signal *s* is generated
-        func will be called
+        """Register *func* to be called when signal *s* is generated.
```

Comment:
It's not obvious to me what the signal `s` is here, or even what type it should have

Reply:
It can indeed be anything (hashable):
```
In [1]: from matplotlib.cbook import CallbackRegistry; cbr = CallbackRegistry()

In [2]: s = object()

In [3]: cbr.connect(s, print)
Out[3]: 1

In [4]: cbr.process(s, "hello world")
hello world
```